C/plugins/filter/python27/plugin.cpp

```diff
@@ -0,0 +1,546 @@
+				"\"type\": \"integer\", " \
+				"\"default\": \"" VALUE_MULTIPLIER "\"}, " \
+			"\"script\" : {\"description\" : \"Python 2.7 module to load.\", " \
+				"\"type\": \"string\", " \
+				"\"default\": \"" SCRIPT_FILE "\"} }"
+using namespace std;
+
+// Python 2.7 loaded filter module handle
+static PyObject* pModule = NULL;
+// Python 2.7 callable method handle
+static PyObject* pFunc = NULL;
+// Python 2.7  script name
+static string pythonScript;
+
+// Filtering methods
+static PyObject* createReadingsList(const vector<Reading *>& readings);
+static vector<Reading *>* getFilteredReadings(PyObject* filteredData);
+static void logErrorMessage();
+
+/**
+ * The Filter plugin interface
+ */
+extern "C" {
+/**
+ * The plugin information structure
+ */
+static PLUGIN_INFORMATION info = {
+        FILTER_NAME,              // Name
+        "1.0.0",                  // Version
+        0,                        // Flags
+        PLUGIN_TYPE_FILTER,       // Type
+        "1.0.0",                  // Interface version
+	DEFAULT_CONFIG	          // Default plugin configuration
+};
+
+/**
+ * Return the information about this plugin
+ */
+PLUGIN_INFORMATION *plugin_info()
+{
+	return &info;
+}
+
+/**
+ * Initialise the plugin, called to get the plugin handle and setup the
+ * output handle that will be passed to the output stream. The output stream
+ * is merely a function pointer that is called with the output handle and
+ * the new set of readings generated by the plugin.
+ *     (*output)(outHandle, readings);
+ * Note that the plugin may not call the output stream if the result of
+ * the filtering is that no readings are to be sent onwards in the chain.
+ * This allows the plugin to discard data or to buffer it for aggregation
+ * with data that follows in subsequent calls
+ *
+ * @param config	The configuration category for the filter
+ * @param outHandle	A handle that will be passed to the output stream
+ * @param output	The output stream (function pointer) to which data is passed
+ * @return		An opaque handle that is used in all subsequent calls to the plugin
+ */
+PLUGIN_HANDLE plugin_init(ConfigCategory* config,
+			  OUTPUT_HANDLE *outHandle,
+			  OUTPUT_STREAM output)
+{
+	FogLampFilter* handle = new FogLampFilter(FILTER_NAME,
+						  *config,
+						  outHandle,
+						  output);
+
+	// Embedded Python 2.7 program name
+        Py_SetProgramName((char *)config->getName().c_str());
+	// Embedded Python 2.7 initialisation
+        Py_Initialize();
+
+	const char* rootDir = getenv("FOGLAMP_ROOT");
+	const char* dataDir = getenv("FOGLAMP_DATA");
+	string filtersPath;
+	if (!dataDir)
+	{
+		filtersPath = (rootDir == NULL ? FOGLAMP_ROOT_PATH : rootDir);
+		filtersPath += "/data";
+	}
+	else
+	{
+		filtersPath = dataDir;
+	}
+	filtersPath += PYTHON_FILTERS_PATH;
+
+	// Set Python path for embedded Python 2.7
+	// Get current sys.path
+	PyObject* sysPath = PySys_GetObject((char *)string("path").c_str());
```

Comment:
Where do you decrement the reference count on these objects?

Reply:
PySys_GetObject returns a borrowed reference. No need to do deal with it